src/tyr/route_serializer_osrm.cc

```diff
@@ -1442,11 +1592,27 @@ json::ArrayPtr serialize_legs(const google::protobuf::RepeatedPtrField<valhalla:
         step->emplace("rotary_name", maneuver.street_name(0).value());
       }
 
+      // Get incoming and outgoing bearing. For the incoming heading, use the
+      // prior edge from the TripLeg. Compute turn modifier. TODO - reconcile
+      // turn degrees between Valhalla and OSRM
+      uint32_t idx = maneuver.begin_path_index();
+      uint32_t in_brg = (idx > 0) ? etp.GetPrevEdge(idx)->end_heading() : 0;
+      uint32_t out_brg = maneuver.begin_heading();
```

Comment:
ok this got pulled out because the function we call now doesnt have the path so it cant look at the previous edge. kind of a mess but im not sure how to simplify it. i understand you wanted to not have one large `osrm_maneuver` method but it does kind of spread logic around. no real clean way to do it, either one large method or smaller ones but with state spread around between them :frowning_face:

Reply:
Yes. And also: both the `turn_modifier` and `osrm_maneuver` function utilise the bearing variables `in_brg` and `out_brg`. The `turn_modifier` was pulled out because it is being used in the `maneuver_type` in the `banner_instructions` function itself (along with the `modifier` from the `turn_modifier` call). 

I did all of that in order to **not** run the same code again when creating the `banner_instructions`.